MCprep_addon/materials/generate.py

```diff
@@ -30,10 +30,18 @@
 
 AnimatedTex = Dict[str, int]
 
-class PackFormat(Enum):
-	SIMPLE = 0
-	SEUS = 1
-	SPECULAR = 2
+class DocEnum(Enum):
+	def __new__(cls, value, doc=""):
+		self = object.__new__(cls)  # calling super().__new__(value) here would fail
+		self._value_ = value
+		if doc != "":
+			self.__doc__ = doc
+		return self
+    
+class PackFormat(DocEnum):
+	SIMPLE = 0, "Use a simple shader setup with no PBR or emission falloff."
+	SEUS = 1, "Sets the pack format to Specular."
+	SPECULAR = 2, "Sets the pack format to SEUS."
```

Comment:
I do agree that this obfuscates the code a bit too much - fun to know this is possible, but I think we can get far enough by just defining the shared settings class.

Reply:
I don't really think it even makes sense to abstract it in the first place. `bpy.props.EnumProperty` is tuple based, so we'll have to do conversions anyway. I think the current way it's done on `master` is already a good compromise between readability and verbosity; any other route is more significantly more verbose for little (if any) readability gains.

The only possible route I can think of that somewhat makes sense is setting each enum in `PackFormat` to be equal to their `bpy.props.EnumProperty` counterpart, but that's not a good idea (creation of unnecessary objects in if statements would be an issue)